src/model/document.ts

```diff
@@ -50,6 +50,29 @@ export class ScannedDocument {
   }
 }
 
+// A map between kind string literal types and their feature types.
+export interface FeatureKinds {
```

Comment:
Strong agreement. I experimented with this a bit but didn't get interface merging working right, and I didn't want to spend much time on this.